src/lib.rs

```diff
@@ -38,11 +38,31 @@ const NUM_RETRIES: u32 = 1 << 31;
 const NUM_RAND_CHARS: usize = 12;
 
 impl TempDir {
-    /// Attempts to make a temporary directory inside of `tmpdir` whose name
+    /// Creates a `TempDir` inside of `tmpdir` whose name
```

Comment:
Any reason you changed the wording from 'attempts to create' to 'creates'? As it returns a `Result`, this can actually fail.